ffs/integrationtest/integration_test.go

```diff
@@ -1242,6 +1242,40 @@ func TestLogHistory(t *testing.T) {
 	require.Greater(t, len(lgs), 3) // Ask to have more than 3 log messages.
 }
 
+func TestResumeScheduler(t *testing.T) {
+	t.Parallel()
+
+	ds := tests.NewTxMapDatastore()
+	ipfs, ipfsMAddr := createIPFS(t)
+	addr, client, ms := newDevnet(t, 1, ipfsMAddr)
+	manager, closeManager := newFFSManager(t, ds, client, addr, ms, ipfs)
+	_, auth, err := manager.Create(context.Background())
+	require.NoError(t, err)
+	time.Sleep(time.Second * 3) // Wait for funding txn to finish.
+	fapi, err := manager.GetByAuthToken(auth)
+	require.NoError(t, err)
+
+	r := rand.New(rand.NewSource(22))
+	c, _ := addRandomFile(t, r, ipfs)
+	jid, err := fapi.PushConfig(c)
+	require.Nil(t, err)
+
+	time.Sleep(time.Second * 3)
+	ds2, err := ds.Clone()
```

Comment:
I created a `Clone()` new API for the in memory datastore. 
So I completely clone the datastore in an arbitrary point of execution, and then use that cloned database in a newly spinned up FFS. 

That is the only way to have a ""corrupted/interrupted/etc"" state of the application. (Since doing any kind of closing would leave things in an expected way).

Hope makes sense to you!

Reply:
Nice! Could there have been a way to leverage `exec` to run the program (after compiling in the test), and then use `Kill` from the command process (`cmd.Process.Kill()`). That might simulate Ctr+C a bit more directly. Just a side thought.

Reply:
👍